infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -183,6 +212,100 @@ public static void main(String[] args) {
         addManagementServer();
         router.baseServer.setOptionsMap(opts);
 
+        // Setup SSL if needed
+        Boolean tlsEnabled = (Boolean) opts.get("--enable-tls");
+        if (tlsEnabled) {
+            // Get the key store password
```

Comment:
Any opinions for or against that small refactoring?  Aside from that question, everything else looks good to me.  As an added bonus, when TLS is enabled, ciphertext does indeed appear to be on the wire, and also attempts to have a non-TLS client to talk to a TLS-enabled server will fail.